ansible/idr-searchengine.yml

```diff
@@ -58,82 +60,148 @@
+        bootstrap.memory_lock: "true"
+        network.host: 0.0.0.0
+        cluster.name: searchengine-cluster
+        cluster.initial_master_nodes: "{{ elasticsearchnodes | join(',') }}"
+        http.host: 0.0.0.0
+        #http.port: 9200
+        ES_JAVA_OPTS: "-Xms2g -Xmx2g"
+        ingest.geoip.downloader.enabled: "false"
       networks:
-      - name: searchengine-net
+      - name: "searchengine-net"
+      published_ports:
+        - "9201:9200"
+        - "9301:9300"
+      state: started
+      restart_policy: always
+      volumes:
+      - "{{ apps_folder }}/searchengine/elasticsearch/node1/data:/var/lib/elasticsearch"
+      - "{{ apps_folder }}/searchengine/elasticsearch/node1/logs:/var/log/elasticsearch"
+      - "{{ elasticsearch_backup_folder }}:{{ elasticsearch_backup_folder }}"
+
+  - name: Run docker elasticsearch for the remaining nodes
+    become: yes
+    docker_container:
+      image: "{{ search_engineelasticsearch_docker_image }}"
+      name: "searchengineelasticsearchnode{{ item }}"
+      ulimits:
+        - 'memlock:-1:-1'
+      cleanup: True
+      env:
+        path.data: "/var/lib/elasticsearch"
+        path.logs: "/var/log/elasticsearch"
+        path.repo: "{{ elasticsearch_backup_folder }}"
+        node.name: "searchengineelasticsearchnode{{ item }}"
+        bootstrap.memory_lock: "true"
+        network.host: 0.0.0.0
+        cluster.name: "searchengine-cluster"
+        discovery.seed_hosts: "searchengineelasticsearchnode1"
+        cluster.initial_master_nodes: "{{ elasticsearchnodes | join(',') }}"
+        http.host: 0.0.0.0
+        #http.port: 9200
+        ES_JAVA_OPTS: "-Xms1g -Xmx1g"
+        ingest.geoip.downloader.enabled: "false"
+
+      networks:
+      - name: "searchengine-net"
       published_ports:
-      - "9201:9200"
-      - "9301:9300"
+        - "920{{ item }}:9200"
+        - "930{{ item }}:9300"
       state: started
       restart_policy: always
       volumes:
-      - "{{ apps_folder }}/searchengine/elasticsearch{{ apps_folder }}:/var/lib/elasticsearch"
-      - "{{ apps_folder }}/searchengine/elasticsearch/logs:/var/log/elasticsearch"
+      - "{{ apps_folder }}/searchengine/elasticsearch/node{{ item }}/data:/var/lib/elasticsearch"
+      - "{{ apps_folder }}/searchengine/elasticsearch/node{{ item }}/logs:/var/log/elasticsearch"
+      - "{{ elasticsearch_backup_folder }}:{{ elasticsearch_backup_folder }}"
+    with_sequence: start=2 count={{ noelasticsearchnodes | int -1}}
 
   - name: configure elasticsearch  for docker searchengine
     become: yes
     docker_container:
       image: "{{ searchengine_docker_image }}"
       name: searchengine_elasticsearch
       cleanup: True
-      #auto_remove: yes
-      command: "set_elasticsearch_configuration -e searchengineelasticsearch"
-      #networks:
-      #- name: searchengine-net
-      #published_ports:
-      #- "5577:5577"
+      command: "set_elasticsearch_backup_folder -b {{ elasticsearch_backup_folder }}"
```

Comment:
This command is not in the current search engine Docker image so this PR will need to be coupled with a new PR and a new image version bump, is that correct?

Reply:
Yes, this is correct,  it is included in this pull request:
https://github.com/ome/omero_search_engine/pull/61